framework/src/functions/LinearCombinationFunction.C

```diff
@@ -61,3 +61,21 @@ LinearCombinationFunction::value(Real t, const Point & p)
     val += _w[i] * _f[i]->value(t, p);
   return val;
 }
+
+RealGradient
+LinearCombinationFunction::gradient(Real t, const Point & p)
+{
+  RealGradient g;
+  for (unsigned i = 0; i < _f.size(); ++i)
+    g += _w[i] * _f[i]->gradient(t, p);
+  return g;
+}
+
+RealVectorValue
+LinearCombinationFunction::vectorValue(Real t, const Point & p)
```

Comment:
Well that's a no go. I'm going to see if @friedmud or @andrsd has an opinion. I really don't like the fact that we have these two, "almost the same" interfaces hanging around in Function IC that people are starting to override. I'd really like to get rid of one of them, but I know there are people that think we should support both "Gradient" and "VectorValue" variants for type correctness.